### PR TITLE
implement jump searcher algorithm.

### PR DESCRIPTION
**Pull request by beqakd**
Implement one of the most public searcher algorithm.
#145 
- [x] I have performed a self-review of my code
- [x] My code follows the style guidelines of this project
- [x] I have added comments to hard-to-understand areas of my code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I have made corresponding changes to the comments
- [ ] I have made corresponding changes to the README.md

### CODE REVIEW

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- Algorithms/Search/JumpSearcher.cs  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5684457&bid=18606286)

File: Algorithms/Search/JumpSearcher.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    /// Note: the array has to be sorted beforehand.
+    /// </summary>
+    public class JumpSearch {
+        /// <summary>
+        /// Code is really easy we find step size first
+        /// Later we find the block where this element should be.!
+        /// In this block we go search for value.! If found return index
+        /// Otherwise return -1!
+        /// </summary>
+        private int jumpSearch (int[] arr, int val) {
```
Comment (siriak) [inline Algorithms/Search/JumpSearcher.cs:18]: Codacy Issue found: [Rename method 'jumpSearch' to match pascal case naming rules, consider using 'JumpSearch'.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5684457)

Comment (beqakd) [reply Algorithms/Search/JumpSearcher.cs:18]: Will change to it!

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Algorithms/Search/JumpSearcher.cs  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5684457&bid=18606286)

**Comment by beqakd** [comment]
Can anyone specify what changes should i make? I was not able to understand why merging is blocked

File: Algorithms/Search/JumpSearcher.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        /// </summary>
+        private int JumpSearch (int[] arr, int val) {
+            double temp = Math.Sqrt (val);
+            int step = (int) Math.Floor (temp);
+
+            int old = 0;
+            while (val > arr[Math.Min (step, arr.Length) - 1]) {
+                old = step;
+                temp = Math.Sqrt (arr.Length);
+                step += (int) Math.Floor (temp);
```
Comment (siriak) [inline Algorithms/Search/JumpSearcher.cs:26]: Modifying step is misleading and this variable isn't named correctly. The step should be fixed and the current position should be updated every iteration. Could you refactor this?

File: Algorithms/Search/JumpSearcher.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+                if (arr.Length <= old) {
+                    return -1;
+                }
+            }
+
+            while (val > arr[old]) {
+                old++;
+                if (old == Math.Min (step, arr.Length)) {
```
Comment (siriak) [inline Algorithms/Search/JumpSearcher.cs:34]: Same here, try to think of better names for all the variables

File: Algorithms/Search/JumpSearcher.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+namespace Algorithms.Search {
+
+    /// <summary>
+    /// The idea: This is one of the public search algorithm.
+    /// It takes 2 argument array of numbers and value of number
+    /// It finds index of this value in given array
+    ///     Time compl is O(root(n))
+    /// Note: the array has to be sorted beforehand.
+    /// </summary>
+    public class JumpSearcher {
```
Comment (siriak) [inline Algorithms/Search/JumpSearcher.cs:11]: Take a look at https://github.com/TheAlgorithms/C-Sharp/blob/master/Algorithms/Search/BinarySearcher.cs
Could this searcher be implemented to work with IComparable<T>?

**Comment by beqakd** [comment]
Thanks a lot for the review! Will do changes right away.

File: Algorithms/Search/JumpSearcher.cs (hunk 4 above)
Comment (beqakd) [reply Algorithms/Search/JumpSearcher.cs:11]: Good point. I totally forgot to implement it.

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Algorithms/Search/JumpSearcher.cs  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5684457&bid=18606286)

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Algorithms/Search/JumpSearcher.cs  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5684457&bid=18606286)

**Comment by siriak** [comment]
Closing because build fails, there are no tests, and code style isn't followed.

**Pull request closed without merging**